content/blog/2022/2022-03-31-be-informed-with-lombok.md

```diff
@@ -0,0 +1,474 @@
+   {{% image alt="delombok example" src="images/posts/lombok/checkstyle_err.png" %}}
+4. With **`@AllArgsConstructor`**  referring to multiple same-type parameters, it is easy to accidentally define parameters out of order. 
+It affects code readability and introduces bugs that can be difficult to trace.
+
+In my experience, I have seen huge complex objects having multiple dependencies.
+Developers tend to use these annotations to escape Sonar checks making it difficult to maintain the code.
+
+### Logic within application should not depend on the generated code
+
+Apart from following good programming practices, developers try to generalize features to ensure re-usability. 
+However, these features should **NEVER** depend on the code that Lombok generates. *For instance, consider we create a base feature that uses reflection to create objects. 
+The DTOs use @Builder, and we use the Lombok generated code in it. If someone decides to create new DTOs 
+that use @Builder(setterPrefix = "with"), this could be catastrophic in huge, complex applications*. 
+Since Lombok provides a lot of flexibility in the way objects are created, we should be equally responsible and use them appropriately.
+
+### @SneakyThrows can be evil
+
+Let's first consider this example:
+````java
+   public interface DataProcessor {
+       void dataProcess();
+   }   
+````
+Without @SneakyThrows an implementation of DataProcessor would be like this:
+````java
+   public class FileDataProcessor implements DataProcessor {
+       @Override
+       public void dataProcess() {
+           try {
+               processFile();
+           } catch (IOException e) {
+               e.printStackTrace();
+           }
+       }
+   
+       private void processFile() throws IOException {
+           File file = new File("sample.txt");
+           throw new IOException(); // forcibly throw
+       }
+   }
+````
+With @SneakyThrows the code gets simplified
+````java
+   public class FileDataProcessor implements DataProcessor {
+       @Override
+       @SneakyThrows
+       public void dataProcess() {
+          processFile();
+       }
+   
+       private void processFile() throws IOException {
+           File file = new File("sample.txt");
+           throw new IOException();
+       }
+   }
+````
+As we can see, this **avoids the hassle of catching or throwing checked exceptions**. In other words, it treats a checked exception like an unchecked one.
+This can be useful, especially when writing lambda functions making the code concise and clean. However, since the annotation swallows the checked exception we cannot catch them explicitly. 
+Instead, we must handle the exceptions via a global exception handler.
+Therefore, **use it only when you don't intend to process the code selectively depending on the kind of Exception it throws**.
+**Ensure it is used cautiously and not as an alternative to bypass checked exceptions.**
```

Comment:
The main feature of `SneakyThrows` is to bypass checked exceptions, so I don't understand this sentence.